pkg/server/ptypes/pipeline_ui.go

```diff
@@ -0,0 +1,408 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package ptypes
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/hashicorp/waypoint/internal/pkg/graph"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+)
+
+func UI_PipelineRunTreeFromJobs(jobs []*pb.Job, statusReports []*pb.StatusReport) (*pb.UI_PipelineRunTreeNode, error) {
```

Comment:
This could use a doc comment detailing what this func does

Reply:
Haha, whoops! Yep, I’ll get right on that.